mlflow/tracking/fluent.py

```diff
@@ -1348,8 +1350,13 @@ def _get_experiment_id():
     return (
         _active_experiment_id
         or _get_experiment_id_from_env()
-        or (is_in_databricks_notebook() and get_notebook_id())
         or (is_in_databricks_job() and get_job_type_info() == "NORMAL" and _create_job_experiment())
+        or (
+            is_in_databricks_notebook()
+            and is_in_databricks_repo_notebook()
+            and _create_repo_notebook_experiment()
+        )
+        or (is_in_databricks_notebook() and get_notebook_id())
```

Comment:
This establishes the desired hierarchy; we should always resolve as a job first, if it is in a job.
I moved the regular notebook line down - it wouldn't conflict with jobs, but this way it's clearer.